src/OpenLoco/src/Paint/Paint.cpp

```diff
@@ -1077,21 +1078,28 @@ namespace OpenLoco::Paint
 
         for (auto* ps = (*_paintHead)->basic.nextQuadrantPS; ps != nullptr; ps = ps->nextQuadrantPS)
         {
-            auto* tempPS = ps;
-            auto* nextPS = ps;
-            while (nextPS != nullptr)
+            // Check main paint struct
+            if (isSpriteInteractedWith(getRenderTarget(), ps->imageId, ps->vpPos))
             {
-                ps = nextPS;
-                if (isSpriteInteractedWith(getRenderTarget(), ps->imageId, ps->vpPos))
+                if (isPSSpriteTypeInFilter(ps->type, flags))
                 {
-                    if (isPSSpriteTypeInFilter(ps->type, flags))
+                    info = { *ps };
+                }
+            }
+
+            // Check children paint structs
+            for (const auto* childPs = ps->children; childPs != nullptr; childPs = childPs->children)
+            {
+                if (isSpriteInteractedWith(getRenderTarget(), childPs->imageId, childPs->vpPos))
+                {
+                    if (isPSSpriteTypeInFilter(childPs->type, flags))
                     {
-                        info = { *ps };
+                        info = { *childPs };
                     }
                 }
-                nextPS = ps->children;
             }
 
+            // Check attached to main paint struct
```

Comment:
This is possibly wrong as it should really be main ps or if it had children the last valid child ps. But we do the same thing in paint structs...